src/dataDisplay/Number/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+// https://github.com/gnosis/safe/wiki/How-to-format-amounts
+const DEFAULT_SHOW_SIGN = false;
+
+type FormatOptions = {
+  currency?: string;
+  showSign?: boolean;
+};
+
+type IntlFormatNumberOptions = Intl.NumberFormatOptions & {
+  notation?: 'compact'; // M, B, T
+  signDisplay?: 'always' | 'auto'; // +/-
+};
+
+export const formatGnosisNumber = (
+  value: string,
+  { currency, showSign = DEFAULT_SHOW_SIGN }: FormatOptions = {}
+): string => {
+  const number = +value;
```

Comment:
The whole idea of using strings for numbers is to avoid parsing them into (limited) JS numbers. This should either use the built-in BigInt or the BigNumber.js lib.
At which point one might ask why this component takes a string in the first place.

Reply:
I'll add the ability to pass in numbers, but `BigInt` doesn't work with `Intl.FormatNumber`.

We have an upper limit of `999e12` so we should be fine. The same goes for small numbers, with a limit of `10e-6`. On the off change the number is `> Number.MAX_SAFE_INTEGER` or `< Number.MIN_SAFE_INTEGER`, I'll assign them to the limit

Reply:
You're right, we're rounding them anyway, so precision arithmetic doesn't matter. Accepting both a string and a number is logical, I think.